### PR TITLE
Handle most prereqs of the form 'one subject in' or 'two subjects in' correctly

### LINKED ISSUE #368: Make prerequisites of the form "one subject in X" work correctly.

### PR DESCRIPTION
**Pull request by miriam-rittenberg**
Mostly resolves #368. Some of the things that still don't work are "one subject in film / music" and "one subject in probability", because I'm not sure how to tell whether a class is on probability. The others are of the form "two subject in CMS / History" or something similar, because I think making this work would require large changes to how we handle prerequisites right now. If you have ideas for handling those or for making this code involve fewer long if statements, please add to this pull request.

### CODE REVIEW

File: src/mixins/reqFulfillment.js [hunk 1]
```diff
@@ -51,13 +51,49 @@ export default {
...
           // If the string is a required prereq, it must be manually dismissed
-          splitReq[i] = 'false';
-        } else if ('()/, '.indexOf(splitReq[i]) < 0) {
+          const req = splitReq[i];
+          let idCategory;
+          let numRequired;
+          // the 'ne subject' is on purpose because some say 'One' and some say 'one'
+          if (req.indexOf('ne subject in') >= 0 || req.indexOf('wo subjects in') >= 0) {
+            if (req.indexOf('ne subject in') >= 0) {
+              numRequired = 1;
```
Comment (georgiashay) [inline src/mixins/reqFulfillment.js]: Are there any cases where it's some other number of subjects?  There are [libraries to convert words to numbers](https://github.com/finnfiddle/words-to-numbers) if we want to make this more extendable, although it's always good to keep our node_modules down.

File: src/mixins/reqFulfillment.js [hunk 2]
```diff
@@ -51,13 +51,48 @@ export default {
...
+          // Set other strings (like "permission of instructor") automatically to false
           // If required as an alternative to other prereqs, will not affect fulfillment
           // If the string is a required prereq, it must be manually dismissed
-          splitReq[i] = 'false';
-        } else if ('()/, '.indexOf(splitReq[i]) < 0) {
+          const req = splitReq[i];
+          let idCategory;
+          let numRequired;
+          if (req.toLowerCase().indexOf('one subject in') >= 0) {
+            if (req.indexOf('ne subject in') >= 0) {
```
Comment (georgiashay) [inline src/mixins/reqFulfillment.js]: It would be clearer if this one was also changed to toLowerCase

Comment (miriam-rittenberg) [reply src/mixins/reqFulfillment.js]: Good point, and also did I delete the check for two subjects in? I should put that back in.

Comment (georgiashay) [reply src/mixins/reqFulfillment.js]: Unresolved

Comment (miriam-rittenberg) [reply src/mixins/reqFulfillment.js]: Which part? I changed it to 'one subject in' last week.

Comment (georgiashay) [reply src/mixins/reqFulfillment.js]: I don't see the check for "two subjects in"

Comment (miriam-rittenberg) [reply src/mixins/reqFulfillment.js]: if (lowercaseReq.indexOf('one subject in') >= 0 || lowercaseReq.indexOf('two subjects in') >= 0) {

Comment (georgiashay) [reply src/mixins/reqFulfillment.js]: Ah, must have missed that

**Comment by georgiashay** [comment]
Made sure it adds the little check marks too so we can see the new feature :)

**Pull request merged**